Source/src/WixSharp/CommonTasks.cs

```diff
@@ -503,6 +503,29 @@ static public Project AddDirs(this Project project, params Dir[] items)
             return project;
         }
 
+        /// <summary>
+        /// Adds the IniFile to the Project.
+        /// </summary>
+        /// <param name="project">The project.</param>
+        /// <param name="item">The item.</param>
+        /// <returns></returns>
+        static public Project AddIniFile(this Project project, Dir item)
+        {
+            return project.AddIniFiles(item);
+        }
+
+        /// <summary>
+        /// Adds the IniFile items to the Project.
+        /// </summary>
+        /// <param name="project">The project.</param>
+        /// <param name="items">The items.</param>
+        /// <returns></returns>
+        static public Project AddIniFiles(this Project project, params Dir[] items)
```

Comment:
Here, too.

Reply:
Not a problem. Txs.

----

Recently you have contributed quite a few new features for WixSharp. Therefore I just wanted to provide you some highlight on the implementation strategies associated with adding new features in the most pain-free way. 

Historically, WixSharp implementation approach for developing specific WiX features involved the following steps:
1. _Creating new WixSharp types that represent WIX entities._ 
   a. Creating new classes and enums
   b. Adding new collection member to the `WixSharp.Project` to store new types
   c. Adding extension methods (`WixSharp.CommonTasks`) for modifying `WixSharp.Project` new collection members  
2. _Extending WixSharp.Compiler with_  
  a. XML serialization routines for the new types.
  b. WXS generation routines for the new types serializated XML
  c. WiX compilers adjustments to accommodate the new WixSharp/WiX types
3. _Composing code samples demonstrating the usage of new WixSharp types_

The model above required `WixSharp.Compiler` being fully aware about the nature of the new feature and WixSharp types associated with it. 

Despite being an obvious anti-pattern this approach had its merits as it was an adequate measure for breaking the controversial WiX/MSI authoring paradigm, which is based on MSI tables instead of deployment requirements.

However WixSharp core has matured and the majority of the new features fit the simple "convert to XML" development use-case. Meaning that in most of the cases if you want to add a new WiX feature to WixSharp you only need to perform two steps `#1.a` and `#3`. The rest will be handled by WixSharp compiler automatically.


1. Creating new WixSharp types that represent WIX entities. 
   1.a Creating new classes and enums
       **Needs to be done by the developer. Need to create new class that implements IGenericEntity interface and amark all serializable members with the `XML` attribute** 
   1.b Adding new collection member to the `WixSharp.Project` to store new types
       _No need for development.. WixSharp already has collections for the new types in all expected 'parent' entities- `WixSharp.Project.GenericItems`, `WixSharp.Dir.GenericItems` and `WixSharp.File.GenericItems`_
   1.b Adding extension methods (`WixSharp.CommonTasks`) for modifying `WixSharp.Project` new collection members  
       _Not beneficial/applicable for generic items_
2. Extending WixSharp.Compiler with  
  2.a XML serialization routines for the new types.
      _No need for development. Done automatically by compiler._
  2.b WXS generation routines for the new types serializated XML
      _No need for development. Done automatically by compiler_
  2.c WiX compilers adjustments to accommodate the new WixSharp/WiX types
      _No need for development. Done automatically by compiler._
3. Composing code samples demonstrating the usage of new WixSharp types
   **Needs to be done by the developer.** 


 The perfect example of the streamlined development model is the [EventSource](https://github.com/oleg-shilo/wixsharp/blob/master/Source/src/WixSharp/EventSource.cs) class.